folders.py

```diff
@@ -24,9 +23,12 @@
     '~/Library/Application Support/Spotify/PersistentCache/Storage')
 LINUX_PERSISTENT_CACHE_PATH = os.path.join(
     os.getenv('XDG_CACHE_HOME', '~/.cache'), 'spotify/Storage')
+WINDOWS_PERSISTENT_CACHE_PATH = os.path.join(
+    os.getenv('LOCALAPPDATA'), 'Spotify\\Storage')
```

Comment:
@Nitemice You may want to have a fallback in `os.getenv` if LOCALAPPDATA is not found. Otherwise non-Windows computers will return an error here, as it tries to do `os.path.join(None, ...)`.

Reply:
I don't suppose you have any suggestions for an appropriate default path. I can't think of anything that's going to possibly be a valid path that doesn't use an environment variable.

Also, I was able to work out the path used by Spotify when it's installed from the Windows Store, so if there's a way to detect if a path exists, we could detect which to use.

Reply:
How about rewriting those lines to

```
if sys.platform == "darwin":
    # macOS
    PERSISTENT_CACHE_PATH = (
        "~/Library/Application Support/Spotify/PersistentCache/Storage"
    )
elif sys.platform == "win32":
    # Windows
    PERSISTENT_CACHE_PATH = os.path.join(
        os.getenv("LOCALAPPDATA"), "Spotify", "Storage"
    )
else:
    # Linux
    PERSISTENT_CACHE_PATH = os.path.join(
        os.getenv("XDG_CACHE_HOME", "~/.cache"), "spotify/Storage"
    )
```

If Windows is more involved, you could also write a `get_windows_persistent_cache_path()` helper function and use `os.path.exists`.

That said, maybe we can first commit what we have now, and then do the more involved Windows solution in an upcoming pull request? What do you think?

Reply:
I actually wrote some very similar code after I wrote my comment last night. If you think it's a good idea, I'm happy to finish it off.
But yeah, fair enough. Let's finish this stuff first before going down that path. In which case, what is there left to do before we merge this? Just add a default for `os.getenv('LOCALAPPDATA')`?

Reply:
If we do the `if-else` route like sketched above, I think no default value is needed. I think we can assume that Windows will have the `LOCALAPPDATA` variable set. The earlier suggestion was for the other platforms to not error, but with the if-else that's no longer needed.

Reply:
So should I go ahead with the `if-else` stuff, or do we want to merge the code as is, or is there something else that need to be done first?

Reply:
`if-else` is needed, so the code doesn't break on Linux or MacOS systems. Otherwise, I think we're good. :)